packages/medusa/src/utils/line-items/retrieve-variants-woth-isolated-product-module.ts

```diff
@@ -0,0 +1,49 @@
+import { ProductVariantDTO } from "@medusajs/types"
+
+/**
+ * Retrieve variants for generating line items when isolated product module flag is on.
+ *
+ * @param remoteQuery
+ * @param variantIds
+ */
+export async function retrieveVariantsWithIsolatedProductModule(
+  remoteQuery,
+  variantIds: string[]
+) {
+  const variantIdsMap = new Map(variantIds.map((id) => [id, true]))
+
+  const variables = {
+    filters: {
+      variants: {
+        id: [...variantIds],
+      },
+    },
+  }
+
+  const query = {
```

Comment:
suggestion(non blocking): Are we sure all the places where this util is used require all the same fields? I believe this function can take `fields` as an arguments which would looks like `["id", "title", ..., "variants.id", ...]` and you can here generate the remote query object using the util we talked about.